protocol/messenger.go

```diff
@@ -4331,11 +4336,69 @@ func (m *Messenger) AllMessageByChatIDWhichMatchTerm(chatID string, searchTerm s
 		return nil, err
 	}
 
-	return m.persistence.AllMessageByChatIDWhichMatchTerm(chatID, searchTerm, caseSensitive)
+	messages, err := m.persistence.AllMessageByChatIDWhichMatchTerm(chatID, searchTerm, caseSensitive)
+	if err != nil {
+		return nil, err
+	}
+
+	return m.filterOutHiddenChatMessages(messages)
+
 }
 
 func (m *Messenger) AllMessagesFromChatsAndCommunitiesWhichMatchTerm(communityIds []string, chatIds []string, searchTerm string, caseSensitive bool) ([]*common.Message, error) {
-	return m.persistence.AllMessagesFromChatsAndCommunitiesWhichMatchTerm(communityIds, chatIds, searchTerm, caseSensitive)
+	messages, err := m.persistence.AllMessagesFromChatsAndCommunitiesWhichMatchTerm(communityIds, chatIds, searchTerm, caseSensitive)
+	if err != nil {
+		return nil, err
+	}
+
+	return m.filterOutHiddenChatMessages(messages)
+}
+
+func (m *Messenger) filterOutHiddenChatMessages(messages []*common.Message) ([]*common.Message, error) {
+	communitiesCache := make(map[string]*communities.Community)
+	chatVisibilityCache := make(map[string]bool)
+	var filteredMessages []*common.Message
+
+	for _, message := range messages {
+		chatVisible, ok := chatVisibilityCache[message.ChatId]
+		if ok && chatVisible {
+			filteredMessages = append(filteredMessages, message)
+			continue
+		}
+
+		chat, ok := m.allChats.Load(message.ChatId)
+		if !ok {
+			return nil, ErrChatNotFoundError
+		}
+
+		if chat.CommunityID == "" {
+			filteredMessages = append(filteredMessages, message)
+			continue
+		}
+
+		community, ok := communitiesCache[chat.CommunityID]
+		if !ok {
+			communityID, err := hexutil.Decode(chat.CommunityID)
+			if err != nil {
+				return nil, err
+			}
+			comm, err := m.communitiesManager.GetByID(communityID)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
```suggestion
			if err != nil {
			        if err == communities.ErrOrgNotFound {
			            continue
			        }
				return nil, err
			}
```

Otherwise, if a community is removed but the message stays, the search will be broken.

Reply:
I'm pretty sure we don't delete the community from the DB, but better safe than sorry